shacl/generate_shacl.py

```diff
@@ -37,7 +37,9 @@ def addDomainShape(propertyName, expectedType):
 # Make shape for expectedRange property
 def addRangeShape(propertyName, expectedType):
     rangeShapeDict[propertyName] = expectedType
-    sh_prop = BNode()
+    sh_prop = BNode(
+        f"RangeShape_{propertyName.split('#')[-1]}_{expectedType.split('#')[-1]}"
```

Comment:
On line 41, you should use the convenience method `bn` rather than the whole `.split` construction